hrm-service/src/contact-job/contact-job-complete.ts

```diff
@@ -73,6 +75,76 @@ const processCompletedContactJob = async (completedJob: CompletedContactJobBody)
   }
 };
 
+export const getAttemptNumber = async (completedJob: CompletedContactJobBody, contactJob: ContactJobRecord) => {
+  // Attempt number can be null when error occurs in lambda before job data is parsed
+  // In this case we need to fetch the job from the database to get the attempt number
+  if (completedJob.attemptNumber != null) {
+    return completedJob.attemptNumber;
+  }
+
+  return  contactJob.numberOfAttempts;
+};
+
+export const getContactJobOrFail = async (completedJob: CompletedContactJobBody) => {
+  const contactJob = await getContactJobById(completedJob.jobId);
+
+  if (!contactJob) {
+    throw new Error(`Could not find contact job with id ${completedJob.jobId}`);
+  }
+
+  return contactJob;
+};
+
+export const handleSuccess = async (completedJob: CompletedContactJobBody) => {
+  if (completedJob.attemptResult !== 'success') return;
```

Comment:
You could use TS to prevent calling this function on places where it's not a guaranteed success, like:
```suggestion
export const handleSuccess = async (completedJob: Omit<CompletedContactJobBody, 'attemptResult'> & { attemptResult: ContactJobAttemptResult.SUCCESS }) => {
```

Same could be done for the failure. Your call if you prefer to leave as it is tho.

Reply:
Mmm... does typescript transpile add a run-time check in your example? Because that is a runtime value that could change based on the message being processed, if I understand the way TS works correctly.

Reply:
No it does not transpiles to runtime check, but it will enforce the callers to perform the check calling this function (like it's done in the actual processor). Anyways it was just an idea, not a required change, your call if you prefer to leave the runtime check inside here too.

Reply:
Ahh I see what you are saying. Since we are checking upstream and this runtime check is only really there to double check that the if statement isn't missing, it *could* be handled in TS.